--- AssmOutput/a-_bisheng.s	2025-12-07 23:57:28.245278200 +0800
+++ AssmOutput/a-_our.s	2025-12-07 23:57:27.482705400 +0800
@@ -1,34 +1,37 @@
 	.text
-	.file	"a-.c"
-	.globl	main                            // -- Begin function main
+	.file	"input.c"
+	.globl	main
 	.p2align	2
 	.type	main,@function
-main:                                   // @main
+main:
 	.cfi_startproc
-// %bb.0:
-	sub	sp, sp, #32
-	.cfi_def_cfa_offset 32
-	str	wzr, [sp, #28]
-	mov	w8, #1                          // =0x1
-	str	w8, [sp, #24]
-	mov	w8, #2                          // =0x2
-	str	w8, [sp, #20]
-	ldr	w8, [sp, #24]
-	ldr	w9, [sp, #20]
-	add	w8, w8, w9
-	add	w8, w8, #3
-	subs	w8, w8, #1
-	str	w8, [sp, #16]
-	ldr	w8, [sp, #16]
-	str	w8, [sp, #12]
-	ldr	w0, [sp, #16]
-	add	sp, sp, #32
-	.cfi_def_cfa_offset 0
+	stp	x29, x30, [sp, #-16]!
+	mov	x29, sp
+	sub	sp, sp, #16
+	mov	w8, #1
+	str	w8, [sp, #4]
+	mov	w8, #2
+	str	w8, [sp, #8]
+	ldr	w13, [sp, #4]
+	ldr	w14, [sp, #8]
+	mov	w10, w13
+	mov	w11, w14
+	add	w15, w10, w11
+	mov	w10, w15
+	mov	w11, #3
+	add	w16, w10, w11
+	mov	w10, w16
+	mov	w11, #1
+	sub	w17, w10, w11
+	str	w17, [sp, #12]
+	ldr	w13, [sp, #12]
+	str	w13, [sp, #16]
+	ldr	w14, [sp, #12]
+	mov	w0, w14
+	add	sp, sp, #16
+	ldp	x29, x30, [sp], #16
 	ret
+	.cfi_endproc
 .Lfunc_end0:
 	.size	main, .Lfunc_end0-main
-	.cfi_endproc
-                                        // -- End function
-	.ident	"BiSheng Enterprise 4.2.0.2.B002 clang version 17.0.6 (2261d9fde4e0)"
-	.section	".note.GNU-stack","",@progbits
-	.addrsig
+
